sources/packages/backend/apps/api/src/services/education-program/education-program.service.ts

```diff
@@ -39,28 +43,38 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
   constructor(
     private readonly dataSource: DataSource,
     private readonly educationProgramOfferingService: EducationProgramOfferingService,
+    private readonly institutionService: InstitutionService,
   ) {
     super(dataSource.getRepository(EducationProgram));
     this.offeringsRepo = dataSource.getRepository(EducationProgramOffering);
   }
 
   /**
-   * Gets a program and ensure that this program belongs
+   * Gets a program and optionally ensures that this program belongs
    * to the expected institution including the institution
    * id in the query.
-   * @param programId Program id.
-   * @param institutionId Expected institution id.
-   * @returns program
+   * @param programId program id.
+   * @param institutionId expected institution id.
+   * @returns education program.
    */
   async getInstitutionProgram(
     programId: number,
-    institutionId: number,
+    institutionId?: number,
   ): Promise<EducationProgram> {
-    return this.repo
-      .createQueryBuilder("programs")
-      .where("programs.id = :programId", { programId })
-      .andWhere("programs.institution.id = :institutionId", { institutionId })
-      .getOne();
+    return this.repo.findOne({
+      select: {
+        id: true,
```

Comment:
Do we need this `id` ? I understand that we need the `institution: { id: true },` selected because the typeorm won't select the required data correctly.

Reply:
This is intended to be a generic method to be extended and the ID from the main entity would be necessary.
I am ok to remove it but I do not see the great benefit for this one.
Also this method was already refactored during this PR from selecting everything to select just 3 columns now 😉